app/src/main/java/io/homeassistant/companion/android/settings/qs/ManageTilesFragment.kt

```diff
@@ -66,143 +81,17 @@ class ManageTilesFragment constructor(
-                        findPreference<Preference>("tile_missing_entity")?.isVisible = entityList.isEmpty()
-                        tileSavePref?.isEnabled = !tileEntity.isNullOrEmpty() && !tileLabel.isNullOrEmpty()
-                    }
-                }
-            }
-            tileId = newValue.toString()
-            return@setOnPreferenceChangeListener true
-        }
-
-        findPreference<Preference>("tile_icon")?.let {
-            it.setOnPreferenceClickListener {
-                iconDialog.show(childFragmentManager, tileId)
-                return@setOnPreferenceClickListener true
-            }
-        }
-
-        findPreference<EditTextPreference>("tile_label")?.setOnPreferenceChangeListener { _, newValue ->
-            tileLabel = newValue.toString()
-            tileSavePref?.isEnabled = !tileEntity.isNullOrEmpty() && !tileLabel.isNullOrEmpty()
-            return@setOnPreferenceChangeListener true
-        }
-
-        findPreference<EditTextPreference>("tile_subtitle")?.let {
-            it.isVisible = Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q
-            it.setOnPreferenceChangeListener { _, newValue ->
-                tileSubtitle = newValue.toString()
-                tileSavePref?.isEnabled = !tileEntity.isNullOrEmpty() && !tileLabel.isNullOrEmpty()
-                return@setOnPreferenceChangeListener true
-            }
-        }
-
-        findPreference<ListPreference>("tile_entity")?.setOnPreferenceChangeListener { _, newValue ->
-            tileEntity = newValue.toString()
-            tileSavePref?.isEnabled = !tileEntity.isNullOrEmpty() && !tileLabel.isNullOrEmpty()
-            return@setOnPreferenceChangeListener true
-        }
+    }
 
-        tileSavePref?.setOnPreferenceClickListener {
-            Log.d(TAG, "Saving data for tile $tileId")
-            val hasTile = tileDao.get(tileId!!)
-            var id = 0
-            if (hasTile != null && !tileList.isNullOrEmpty()) {
-                for (item in tileList!!) {
-                    if (item.id == hasTile.id)
-                        id = hasTile.id
-                }
-            }
-            val tileData = TileEntity(
-                id,
-                tileId!!,
-                findPreference<Preference>("tile_icon")?.summary.toString().toIntOrNull(),
-                tileEntity.toString(),
-                tileLabel.toString(),
-                tileSubtitle
-            )
-            tileDao.add(tileData)
-            tileList = tileDao.getAll()
-            Toast.makeText(requireContext(), commonR.string.tile_updated, Toast.LENGTH_SHORT).show()
-            return@setOnPreferenceClickListener true
+    private fun reloadIcon() {
+        val iconDrawable = viewModel.selectedIcon.value?.let { iconPack.getIcon(it)?.drawable }
+        if (iconDrawable != null) {
+            val icon = DrawableCompat.wrap(iconDrawable)
+            icon.setColorFilter(resources.getColor(R.color.colorAccent), PorterDuff.Mode.SRC_IN)
```

Comment:
No longer required because you're doing it in Compose now (2a982aca97dd4a7bdfe5f541ab4f9dd05f87874d)